File: api/src/main/java/dev/obscuria/fragmentum/server/FragmentumServerRegistry.java
package dev.obscuria.fragmentum.server;

import com.mojang.brigadier.CommandDispatcher;
import net.minecraft.commands.CommandBuildContext;
import net.minecraft.commands.CommandSourceStack;
import net.minecraft.commands.Commands;
import org.apache.commons.lang3.NotImplementedException;

@SuppressWarnings("unused")
public interface FragmentumServerRegistry
{
    static void registerCommand(CommandRegistrar registrar)
    {
        throw new NotImplementedException();
    }

    @FunctionalInterface
    interface CommandRegistrar
    {
        void register(
                CommandDispatcher<CommandSourceStack> dispatcher,
                CommandBuildContext registryAccess,
                Commands.CommandSelection environment);
    }
}

File: api/src/main/kotlin/dev/obscuria/fragmentum/server/FragmentumServerRegistry.kt
@file:Suppress("unused", "UNUSED_PARAMETER", "RedundantNullableReturnType")

package dev.obscuria.fragmentum.server

import com.mojang.brigadier.CommandDispatcher
import net.minecraft.commands.CommandBuildContext
import net.minecraft.commands.CommandSourceStack
import net.minecraft.commands.Commands

object FragmentumServerRegistry {

    fun registerCommand(registrar: CommandRegistrar): Unit = error("API stub")

    fun interface CommandRegistrar {

        fun register(
            dispatcher: CommandDispatcher<CommandSourceStack>,
            registryAccess: CommandBuildContext,
            environment: Commands.CommandSelection
        )
    }
}